gravitee-apim-console-webui/src/services-ngx/constants.service.ts

```diff
@@ -72,4 +72,22 @@ export class ConstantsService {
       return matchedSecurityType.length > 0 && matchedSecurityType[0].enabled;
     });
   }
+
+  getPlanMenuItems(definitionVersion: DefinitionVersion, listenerTypes: ListenerType[]): PlanMenuItemVM[] {
+    const availablePlanMenuItems = this.getEnabledPlanMenuItems();
+
+    if (definitionVersion === 'V4' && listenerTypes && listenerTypes.every((listenerType) => listenerType === 'TCP')) {
```

Comment:
it is not a blocker, but those "if" need a bit of effort to understand, we can maybe extract them in meaningful private method ?